nugets/Devolutions.NowProto/INowDeserialize.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace Devolutions.NowProto
+{
+    /// <summary>
+    /// Deserializable NowProto message.
+    /// </summary>
+    public interface INowDeserialize<out T> : INowMessage
```

Comment:
note: It’s actually `T` that must implements the `INowMessage` interface, but I understand that T and the class implementing `INowDeserialize` must be the same type. I wonder if there is a better way of writing this code. I believe C#’s type system is not strong enough.

suggestion: However, I more proper way of going about this kind of stuff that I saw in the standard library, was to require two different types like for the `JsonConverter` interface:

https://github.com/Devolutions/devolutions-gateway/blob/master/utils/dotnet/Devolutions.Gateway.Utils/src/ApplicationProtocolJsonConverter.cs